### ISSUE TITLE
Incorrect "Comparison" value for "Time" stat in End of Run panel

### ISSUE THREAD

**Issue by cesarila** [momentum-mod/game#2222]
### Describe the bug

When getting a personal best on a map I've zoned myself, the comparison column is to a run I've never gotten. I'm guessing that the margin I've gotten a personal best by is being subtracted from my new personal best for the "Comparison" column. For example, if I improve my personal best from 25.00 to 24.80 on a map, the time in the "Comparison column" will be 24.60 seconds instead of 25.00 seconds.

Similarly, when I fail to get a personal best, the margin that I was behind PB by gets added to my current attempt time when displaying the time in the "Comparison" column. For example, if my PB is 20 seconds, and I complete a 30 second run, the time in the "Comparison column will be 40 seconds.

Run Comparison Screen:
![image](https://github.com/user-attachments/assets/bf4b197b-4470-4d13-87e8-f0c1d464447f)
Local Leaderboards:
![image](https://github.com/user-attachments/assets/8b355a6e-102c-40af-b5c9-4e461fdaca1a)

### How To Reproduce

1. Open any bhop map and set a personal best if one does not exist.
2. Complete the map with a different time than the personal best.
3. View the time listed in the "Comparison" column.

Demonstration Video (timestamps in description): https://youtu.be/MLQNearbrMw

### Issue Map

Systematic Issue. I tested on bhop_bobop, which I zoned myself. The same issue occurs on bhop_eazy and bhop_everfast, which I downloaded through momentum mod.

### Expected Behavior

When I complete a map slower than my personal best, my personal best time should be listed in the "Comparison" column. When I get a personal best, my previous personal best time should be listed in the "Comparison" column.

### Operating System

Windows 10 Home 22H2 Build 19045.5131

**Comment by tsa96** [comment]
Yeah this just a case of a value being added when it should've been subtracted. Problem exists on both 0.9 and 0.10 branches, I'll PR a fix into 0.10 shortly, not bothering with 0.9.
